subsys/net/ip/net_context.c

```diff
@@ -746,6 +746,50 @@ int net_context_put(struct net_context *context)
 	return ret;
 }
 
+int net_context_put_all(struct net_if *iface)
+{
+	int i;
+	int ret = 0;
+
+	k_sem_take(&contexts_lock, K_FOREVER);
+
+	for (i = 0; i < NET_MAX_CONTEXT; i++) {
+		if (!net_context_is_used(&contexts[i])) {
+			continue;
+		}
+
+		if (net_context_get_iface(&contexts[i]) != iface) {
+			continue;
+		}
+
+		k_mutex_lock(&contexts[i].lock, K_FOREVER);
+
+		if (IS_ENABLED(CONFIG_NET_OFFLOAD) &&
+		    net_if_is_ip_offloaded(iface)) {
+			contexts[i].flags &= ~NET_CONTEXT_IN_USE;
+			ret = net_offload_put(iface, &contexts[i]);
+			k_mutex_unlock(&contexts[i].lock);
+			continue;
+		}
+
+		contexts[i].connect_cb = NULL;
+		contexts[i].recv_cb = NULL;
+		contexts[i].send_cb = NULL;
+
+		/* net_tcp_put() will handle decrementing refcount on stack's behalf */
+		net_tcp_put(&contexts[i], true);
+
+		/* Decrement refcount on user app's behalf */
+		net_context_unref(&contexts[i]);
```

Comment:
But I'm not sure if we should dereference net context as well? On the application side there's still an open socket that needs to be closed, when we close the socket the context would be unreffed by the socket code again I think?

Probably better just to have the net stack to report an error on the socket, and let the application close it. The question is, whether we should do it only for the sockets with active connections (i.e. TCP only, which should be handled by `net_tcp_put()` call here already), or really all sockets (we'd need some extra handling for UDP sockets then). IMHO, TCP only is fine, I don't think there's a need to close connection-less sockets?